featurebyte/models/target.py

```diff
@@ -33,17 +33,17 @@ class TargetModel(FeatureByteCatalogBaseDocumentModel):
     """
 
     # Recipe
-    graph: QueryGraph = Field(allow_mutation=False)
-    node_name: str
+    graph: Optional[QueryGraph] = Field(allow_mutation=False)
+    node_name: Optional[str] = Field(allow_mutation=False)
```

Comment:
Maybe we should move the `Field(allow_mutation=False)` to the api object. The reason that you may find it in our model file is because previously we inherit it from model. For our current api object, we try not to inherit document model directly.

Reply:
hm ok, so no `allow_mutation`s in the models?

Reply:
It is ok to keep it in the models. The main propose is to make it immutable for api object.